keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -75,6 +77,44 @@ class ArtifactListener(
       }
   }
 
+  /**
+   * Finds the delivery configs that are using an artifact,
+   * and publishes a build lifecycle event for them.
+   */
+  fun createBuildLifecycleEvent(artifact: PublishedArtifact) {
+    log.debug("Publishing build lifecycle event for artifact $artifact")
+    artifact.buildMetadata
+      ?.let { buildMetadata ->
+
+        val data: MutableMap<String,String?> = mutableMapOf()
+        data["buildNumber"] = artifact.metadata["buildNumber"]?.toString()
+        data["commitId"] = artifact.metadata["commitId"]?.toString()
+        data["fallbackLink"] = buildMetadata.job?.link
```

Comment:
```suggestion
        val data = mutableMapOf(
          "buildNumber" to artifact.metadata["buildNumber"]?.toString(),
          "commitId" to artifact.metadata["commitId"]?.toString(),
          "fallbackLink" to buildMetadata.job?.link
        )
```

Reply:
Why does it need to be mutable?

Reply:
Because we add application in it for each delivery artifact (line 103). But I'll change to the more kotlin-y way of initializing. i was apparently stuck in java land